model/src/main/kotlin/VcsType.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Copyright (C) 2019 Bosch Software Innovations GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.model
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonValue
+
+/**
+ * An class for Version Control System types. Each type has one or more [aliases] associated to it, where the first
+ * alias is the definite name.
+ */
+data class VcsType(val aliases: List<String>) {
+    @JsonCreator
+    constructor(type: String): this(
+        ALL_ALIASES.find { aliases ->
+            aliases.find { alias ->
+                alias.equals(type, true)
+            } != null
+        } ?: listOf(type)
+    )
+
+    companion object {
+        /**
+         * [Git](https://git-scm.com/) - the stupid content tracker.
+         */
+        val GIT = VcsType(listOf("Git"))
+
+        /**
+         * [Repo](https://source.android.com/setup/develop/repo) complements Git by simplifying work across multiple
+         * repositories.
+         */
+        val GIT_REPO = VcsType(listOf("GitRepo", "git-repo", "repo"))
+
+        /**
+         * [Mercurial](https://www.mercurial-scm.org/) is a free, distributed source control management tool.
+         */
+        val MERCURIAL = VcsType(listOf("Mercurial", "hg"))
+
+        /**
+         * [Subversion](https://subversion.apache.org/) is an open source Version Control System.
+         */
+        val SUBVERSION = VcsType(listOf("Subversion", "svn"))
+
+        /**
+         * The [Concurrent Versions System](https://en.wikipedia.org/wiki/Concurrent_Versions_System), not actively
+         * developed anymore.
+         */
+        val CVS = VcsType(listOf("CVS"))
+
+        private val ALL_ALIASES = listOf(
+            GIT.aliases,
+            GIT_REPO.aliases,
+            MERCURIAL.aliases,
+            SUBVERSION.aliases,
+            CVS.aliases
+        )
+
+        /**
+         * An unknown VCS type.
+         */
+        val UNKNOWN = VcsType(listOf(""))
```

Comment:
nit: Did you consider using none instead?

Reply:
Actually given the argumentation that we use class instead of enum means that there can be VCS types in the model which are not known to ORT. From that perspective assigning them to an empty string seems wrong to me as it could be any value. 

Can we just remove this unknown constant ?

Reply:
You're right, this should be `NONE`, as `UNKNOWN` should be reserved for any type / string that is not in `ALL_ALIASES`.